satpy/readers/seviri_l1b_nc.py

```diff
@@ -60,29 +61,46 @@ def end_time(self):
         """Get the end time."""
         return self.deltaEnd
 
-    def _read_file(self):
+    @cached_property
+    def nc(self):
         """Read the file."""
-        if self.nc is None:
+        f_obj = open_file_or_filename(self.filename)
 
-            self.nc = xr.open_dataset(self.filename,
-                                      decode_cf=True,
-                                      mask_and_scale=False,
-                                      chunks=CHUNK_SIZE)
+        dataset = xr.open_dataset(f_obj,
+                                  decode_cf=True,
+                                  mask_and_scale=False,
+                                  chunks=CHUNK_SIZE)
 
+        return dataset.rename({'num_columns_vis_ir': 'x', 'num_rows_vis_ir': 'y'})
+
+    def get_metadata(self):
+        """Get metadata."""
         # Obtain some area definition attributes
-        equatorial_radius = (self.nc.attrs['equatorial_radius'] * 1000.)
+        equatorial_radius = self.nc.attrs['equatorial_radius'] * 1000.
         polar_radius = (self.nc.attrs['north_polar_radius'] * 1000 + self.nc.attrs['south_polar_radius'] * 1000) * 0.5
         ssp_lon = self.nc.attrs['longitude_of_SSP']
+        self.mda['vis_ir_grid_origin'] = self.nc.attrs['vis_ir_grid_origin']
+        self.mda['vis_ir_column_dir_grid_step'] = self.nc.attrs['vis_ir_column_dir_grid_step'] * 1000.0
+        self.mda['vis_ir_line_dir_grid_step'] = self.nc.attrs['vis_ir_line_dir_grid_step'] * 1000.0
+        # if FSFile is used h5netcdf engine is used which outputs arrays instead of floats for attributes
+        if isinstance(equatorial_radius, np.ndarray):
+            equatorial_radius = equatorial_radius[0]
+            polar_radius = polar_radius[0]
+            ssp_lon = ssp_lon[0]
+            self.mda['vis_ir_column_dir_grid_step'] = self.mda['vis_ir_column_dir_grid_step'][0]
+            self.mda['vis_ir_line_dir_grid_step'] = self.mda['vis_ir_line_dir_grid_step'][0]
```

Comment:
I think `item` would suit better here. Also maybe this can be delegated to a small method (including the `if` statement)

```suggestion
            equatorial_radius = equatorial_radius[0]
            polar_radius = polar_radius.item()
            ssp_lon = ssp_lon.item()
            self.mda['vis_ir_column_dir_grid_step'] = self.mda['vis_ir_column_dir_grid_step'].item()
            self.mda['vis_ir_line_dir_grid_step'] = self.mda['vis_ir_line_dir_grid_step'].item()
```

Reply:
Will refactor to item. I was thinking about a method too but I am not sure if it is worth the overhead?

Reply:
I wouldn't bother to put `item` in a special method.